pywt/setup.py

```diff
@@ -10,20 +10,43 @@ def configuration(parent_package='', top_path=None):
 
     config.add_data_dir('tests')
 
-    sources = ["_pywt", "common", "convolution", "wavelets", "wt"]
-    source_templates = ["convolution", "wt"]
-    headers = ["templating", "wavelets_coeffs"]
-    header_templates = ["convolution", "wt", "wavelets_coeffs"]
+    sources = ["c/common", "c/convolution", "c/wavelets", "c/wt"]
+    source_templates = ["c/convolution", "c/wt"]
+    headers = ["c/templating", "c/wavelets_coeffs"]
+    header_templates = ["c/convolution", "c/wt", "c/wavelets_coeffs"]
+
+    c_files = ["_extensions/{0}.c".format(s) for s in sources]
+    depends = (["_extensions/{0}.template.c".format(s) for s in source_templates]
+               + ["_extensions/{0}.template.h".format(s) for s in header_templates]
+               + ["_extensions/{0}.h".format(s) for s in headers]
+               + ["_extensions/{0}.h".format(s) for s in sources])
+
+    config.add_subpackage(
+        '_extensions',
+        subpackage_path="_extensions",
```

Comment:
Now reading this properly (it's late):

> Also can somebody familiar with numpy distutils chime in whether I've done the right thing with pywt/setup.py - adding _extensions as a subpackage to get the __init__.py included?

I see why you need this now. ``add_subpackage`` does the job, but it would be a bit cleaner to add ``_extensions/setup.py`` as I mentioned above.